kernel/Kconfig.smp

```diff
@@ -126,4 +126,15 @@ config TICKET_SPINLOCKS
 	  which resolves such unfairness issue at the cost of slightly
 	  increased memory footprint.
 
+config CURRENT_CPU_VALIDATE
+	bool "Validate usage of _current_cpu"
+	depends on SMP && ASSERT
+	default y
```

Comment:
The assertion is compiled when `CONFIG_SMP=y`, this Kconfig is to simply compile out that assertion completely so I guess the number of CPUs shouldn't matter

Reply:
Had another look in the morning and I get what you meant now - the test is necessary only when there's more than 1 CPUs, and can be disabled otherwise.

Reply:
Another thought - maybe we should just change the `CONFIG_SMP` here to `CONFIG_MP_MAX_NUM_CPUS > 1`?

https://github.com/zephyrproject-rtos/zephyr/blob/67980643d799f76aff93118346a8dabfb472a642/include/zephyr/kernel_structs.h#L254-L268